src/_utils/keycodes.ts

```diff
@@ -4,4 +4,25 @@ export const KEYCODES = {
   SPACEBAR: 32,
 }
 
+export const KEY = {
+  ESCAPE: 'Escape',
+  ENTER: 'Enter',
+  SPACEBAR: ' ',
+  TAB: 'Tab',
+  ARROWDOWN: 'ArrowDown',
+  ARROWLEFT: 'ArrowLeft',
+  ARROWRIGHT: 'ArrowRight',
+  ARROWUP: 'ArrowUp',
+}
+
+export const KEYS_TRIGGERING_KEYBOARD_NAVIGATION = [
+  KEY.TAB,
+  KEY.ARROWDOWN,
+  KEY.ARROWLEFT,
+  KEY.ARROWRIGHT,
+  KEY.ARROWUP,
+]
+
+export const KEYS_TRIGGERING_KEYBOARD_NAVIGATION_COMPLEMENTARY = [KEY.ENTER, KEY.SPACEBAR]
```

Comment:
I don't understand why this is a separate variable and not included in `KEYS_TRIGGERING_KEYBOARD_NAVIGATION`

Reply:
I do it this way because ENTER and SPACEBAR are related to keyboard navigation if not on `textarea` or `input[type=text]`. That's why I'm checking the event.target.tagName in the provider (`onKeyDown` function)

Reply:
And I finally moved this logic into the `onKeyDown` function by filtering the array.